web/src/emoji.ts

```diff
@@ -211,21 +276,21 @@ export function update_emojis(realm_emojis) {
 
 // This function will provide required parameters that would
 // need by template to render an emoji.
-export function get_emoji_details_by_name(emoji_name) {
+export function get_emoji_details_by_name(emoji_name: string): EmojiInfo {
     // To call this function you must pass an emoji name.
     if (!emoji_name) {
         throw new Error("Emoji name must be passed.");
     }
 
-    const emoji_info = {emoji_name};
+    const emoji_info: EmojiInfo = {emoji_name, reaction_type: "", emoji_code: ""};
```

Comment:
This feels pretty ugly; maybe you can add a second commit, probably best as a prep commit but it could work do it as a commit on top of this, that just refactors the two blocks below to each create this object at the end and return it, rather than creating this incomplete type here? (I think we should keep that kind of refactor out of the translation commit).